### PR TITLE
Migrates Post to Swift.

### PR DESCRIPTION
**Pull request by diegoreymendez**
Migrates `Post` to Swift.

### Testing:

Several unit tests have been added to try and reduce the chance that this port may fail.

**Test 1:**
Run the unit tests

**Test 2:**
1. Go to the list of posts for a blog.
2. Interact with the posts list: new, edit, view, post, post, restore.
3. Filter by status, search posts.  Make sure everything works.

Needs review: @SergioEstevao

### CODE REVIEW

File: WordPress/WordPressTest/PostTests.swift [hunk 1]
```diff
@@ -71,7 +71,8 @@ class PostTests: XCTestCase {
         post.blog = blog
         post.setCategoriesFromNames(["One", "Three"])
 
-        let postCategories = post.categories as! Set<PostCategory>
+        let postCategories = post.categories!
+        //let postCategories = post.categories as! Set<PostCategory>
```
Comment (SergioEstevao) [inline WordPress/WordPressTest/PostTests.swift]: Commented code? Any reason why?

**Comment by SergioEstevao** [comment]
Beside the comment about commented code on the test, everything looks great. :shipit:

File: WordPress/WordPressTest/PostTests.swift (hunk 1 above)
Comment (diegoreymendez) [inline WordPress/WordPressTest/PostTests.swift]: Just forgot to remove it.  Will do that now.

**Comment by diegoreymendez** [comment]
Thanks Sergio!

**Pull request merged**